### PR TITLE
Upgrade Hudi version to 0.12.2

### LINKED ISSUE #15536: Upgrade hudi version to 0.12.2
https://hudi.apache.org/releases/release-0.12.2/

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #15536

## Release notes

(x) This is not user-visible or docs only and no release notes are required.

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java [hunk 1]
```diff
@@ -76,23 +76,23 @@ public void testCopyOnWriteShowCreateTable()
...
+                                    CREATE TABLE default.%s (
+                                      _hoodie_commit_time STRING,
+                                      _hoodie_commit_seqno STRING,
+                                      _hoodie_record_key STRING,
+                                      _hoodie_partition_path STRING,
+                                      _hoodie_file_name STRING,
+                                      id BIGINT,
+                                      name STRING,
+                                      price INT,
+                                      ts BIGINT)
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java:88]: was this change "required" for the upgrade?

**Review by hashhar — APPROVED** [review]
LGTM

**Review by findinpath — APPROVED** [review]

**Comment by findinpath** [comment]
Related PR https://github.com/trinodb/docker-images/pull/158

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java (hunk 1 above)
Comment (ebyhr) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hudi/TestHudiSparkCompatibility.java:88]: Yes, it comes from the image upgrade.

**Pull request merged**